user-operator/src/main/java/io/strimzi/operator/user/operator/KafkaUserOperator.java

```diff
@@ -174,6 +178,7 @@ public KafkaUserOperator(Vertx vertx,
 
         CompositeFuture.join(
                 scramShaCredentialOperator.reconcile(user.getName(), password),
+                kafkaUserQuotasOperator.reconcile(user.getName(), user.quotasToJson()),
```

Comment:
Are you sure `user.getName()` is the right here? AFAIK the username will be different for TLS user for example and you will need to handle it in the similar way to how it is done for ACLs, or? (unlike passwords, where username always equals name of the CR).

Reply:
I do not think you solved it. You need to handle it as for the ACLs, otherwise you don't cover both possible usernames.

Reply:
Please confirm it is fixed in https://github.com/strimzi/strimzi-kafka-operator/pull/2002/commits/c5770854d3001e3ef5537c25b27e58f077056afe

Reply:
The KafkaUser can exist only in one form - either as TLS use  `CN=user` or as regular user `user`. I guess ideally you need to make sure that the quotas are set for the right one and undet for the other one. I.e. you set the right quotas for the right user but at the same time unset them for the other form. That protects you against situations such as when the user changes the authentication type etc.

I guess for quotas this might be less important than for ACLs, but we might still want to do it to make it clear what we keep int he quota database.